### PR TITLE
feat: add support for spaces search

### PR DESCRIPTION
**Pull request by wa0x6e**
### Summary

Closes: #389 

This PR add the ability to search for a space

### UI/UX changes

- The search input should be pre-populated on page load with the query string from the url (`q=xxx`) if available
- Use different placeholder for space and proposal search
- Explore page now have a message when no spaces results
- Move explore page protocol to query string in the url (`?p=snapshot|snapshotx`)

### New features

- Using the topnav searchbar from the "My" pages should redirect to the explore page, and show a list of spaces matching the search results

### How to test

1. Go to http://localhost:8080/#/home
2. Search for a space using the top searchbar
3. It should show your results in the /explore page
4. You can switch protocol, without to search for the keyword again
5. The URL should reflect your search filters, reloading the page will keep the same results

### TODO

- [x] Code improvement

### CODE REVIEW

**Comment by wa0x6e** [comment]
PR ready UI/UX/Feature wise, will take another look later for code improvement before asking for code review

**Comment by bonustrack** [comment]
Space explore page was working initially then it now stuck for me with this error:
image

**Comment by wa0x6e** [comment]
> Space explore page was working initially then it now stuck for me with this error:
> 
> image
> 
> 

This is because hub is returning null when no results. A pr already open to fix that

**Comment by wa0x6e** [comment]
> Space explore page was working initially then it now stuck for me with this error: image

Should now be fixed

File: apps/ui/src/networks/common/graphqlApi/index.ts [hunk 1]
```diff
@@ -376,14 +376,21 @@ export function createApi(uri: string, networkId: NetworkID, opts: ApiOptions =
       { limit, skip = 0 }: PaginationOpts,
       filter?: SpacesFilter
     ): Promise<Space[]> => {
+      const _filter: Record<string, any> = clone(filter || {});
+
+      if (_filter.searchQuery) {
+        _filter.metadata_ = { name_contains_nocase: _filter.searchQuery };
+      }
+      delete _filter.searchQuery;
```
[the remark this answers is not in the document]
Comment (wa0x6e) [reply apps/ui/src/networks/common/graphqlApi/index.ts:384]: putting it inside will not call the `delete` in case `_filter.searchQuery == ''`

**Pull request merged**